renderdoc/driver/vulkan/vk_acceleration_structure.cpp

```diff
@@ -836,6 +945,162 @@ VulkanAccelerationStructureManager::Allocation VulkanAccelerationStructureManage
   return readbackmem;
 }
 
+VulkanAccelerationStructureManager::Allocation VulkanAccelerationStructureManager::CreateReplayMemory(
+    MemoryType memType, VkDeviceSize size, VkBufferUsageFlags extraUsageFlags)
```

Comment:
Having a separate path for allocating memory during capture for the copied inputs makes sense for now, but adding a new separate memory allocation path here during replay isn't desirable when we already have `AllocateMemoryForResource` for this exact purpose.

As I can see it the only problem with using that right now is that it doesn't allow freeing memory piecewise so anything allocated with all other upload memory would hang around indefinitely. But that shouldn't be an issue if we add a new `MemoryScope` so that it can be freed in batch once initial contents are uploaded the first time.

Reply:
Am I OK to add this in a follow-up patch?

Reply:
I'd prefer do that in this PR as the issue I have is adding unnecessary duplicated code - I'd rather not add it at all, than add then remove it. It should be as simple as adding an enum and using the existing functionality.